lib/BaseClient.ts

```diff
@@ -0,0 +1,25 @@
+import Logger from './Logger';
+
+abstract class BaseClient {
+  logger: Logger;
+  status!: ClientStatus;
+
+  constructor() {
+    this.logger = Logger.global;
+  }
+
+  setStatus(val: ClientStatus): void {
+    this.logger.info(`${this.constructor.name} status: ${val}`);
```

Comment:
instead of `${val}`, rather use `${ClientStatus[val]}` for better logging output (it will give you the enum value, `DISABLED`, instead of the key, `0`)

Reply:
Good point, I overlooked that the enum lost its string value.